### PR TITLE
add Arup specific interface naming

### PR DESCRIPTION
**Pull request by ftschindler**
As adopted in the Development Study Club on 2022-08-11, this PR adds the Arup specific naming convention for interface classes.

### CODE REVIEW

**Comment by ftschindler** [comment]
I can't seem to be able to request a review here, so I'll just ping @nathanrobertsarup, @LloydPickering and @tim-lyon-arup about it.

**Comment by tim-lyon-arup** [comment]
I am also unable to review! @ramaseshankannan are you able to add the `digital-development` group to have write access (and therefore ability to review) to the repo?

**Review by tim-lyon-arup — DISMISSED** [review]

**Comment by ftschindler** [comment]
I'l let it sit for a bit to see if other approvals/request come in.

**Review by ghost — DISMISSED** [review]

File: cppguide.html [hunk 1]
```diff
@@ -178,6 +178,11 @@ <h2 id="Arup_changes">Arup-specific edits</h2>
   <li>
     <a href="#Directory_Names">Directory names</a>: Guidance added.
   </li> 
+    <a href="#Interface_Names">Naming interfaces</a>: interfaces carry the suffix
+    <code>Interface</code>, derived classes carry the suffix of the inteface (i.e.,
+    <code>class ConcreteFoo : public FooInterface {};</code>), regardless of the implementation
+    details of the interface (stativ vs. dynamic vs. using concepts)
```
Comment (tim-lyon) [inline cppguide.html:184]:
```suggestion
    details of the interface (static vs. dynamic vs. using concepts)
```

File: cppguide.html [hunk 2]
```diff
@@ -4347,6 +4352,22 @@ <h3 id="Macro_Names">Macro Names</h3>
...
+
+<p>Postfix interface classes with <code>Interface</code> and
+derived classes with the interface name
+(<code>class ConcreteFoo : public FooInterface {};</code>).</p>
+
+<p>The rationale behind this is that there there are various
+techniques of implementing interfaces in C++ (dynamic polymorphism
+with pure virtual functions, static polymorphism with CRTP,
+static polymorphism with concepts) and we want to have a single
+naming scheme regardless of the implementation detailed (which
```
Comment (tim-lyon) [inline cppguide.html:4365]:
```suggestion
naming scheme regardless of the implementation details (which
```

**Comment by ftschindler** [comment]
Thx for reviving this, @tim-lyon. Lets get this merged, approvals anyone?

**Review by tim-lyon — APPROVED** [review]

**Review by ramaseshankannan — APPROVED** [review]

**Pull request merged**